ecl/hthor/hthor.cpp

```diff
@@ -2402,15 +2420,20 @@ const void * CHThorGroupDedupKeepLeftActivity::nextRowGE(const void * seek, unsi
             break;
         }
 
-        if (numKept < numToKeep-1)
+        if (compareBest)
+        {
+            if (compareBest->docompare(prev,next) < 0 )
```

Comment:
The different comparisons are deliberate.  Sometimes, the row is being skipped over.  Other times, the older row is being replaced by a newer row.  I've check the code login and tests in Thor, HThor and roxie return consistent results.  Please can you re-check and let me know if you still have concerns.